common/src/main/java/com/wynntils/models/abilities/ShamanTotemModel.java

```diff
@@ -71,7 +83,8 @@ public void onTotemSpawn(AddEntityEvent e) {
                 () -> {
                     // didn't come from a cast within the delay, probably not casted by the player
                     // this check needs to be ran with a delay, the cast/spawn order is not guaranteed
-                    if (System.currentTimeMillis() - totemCastTimestamp > CAST_MAX_DELAY_MS) return;
+                    if (!hasAutoCasterItem() && System.currentTimeMillis() - totemCastTimestamp > CAST_MAX_DELAY_MS)
```

Comment:
If you want to make this a "performance fix", maybe you should change order here and allow the time stamp check to short-circuit the test before jumping into the more complex hasAutoCasterItem.